### PR TITLE
[TACHYON-549] General Utilities

### PR DESCRIPTION
**Pull request by Yours-shupeng**
We make `getMasterAddress()` and `getWorkerAddress()` utilities functions which are used in many classes. 
Following the last PR [1032](https://github.com/amplab/tachyon/pull/1032)

### CODE REVIEW

File: servers/src/main/java/tachyon/worker/block/BlockWorkerUtils.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+package tachyon.worker.block;
+
+import java.net.InetSocketAddress;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.util.NetworkUtils;
+
+public final class BlockWorkerUtils {
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/BlockWorkerUtils.java]: Please add a short javadoc describing the class.

File: servers/src/main/java/tachyon/worker/block/BlockWorkerUtils.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  /**
+   * Gets the Tachyon master address from the configuration
+   *
+   * @return the InetSocketAddress of the master
+   */
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/worker/block/BlockWorkerUtils.java]: Please describe the arguments of the method in the java doc (same comment for method below).

File: servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    // initialize the null master host name test properties.
+    sNullMasterHostNameTestProperties.put(Constants.MASTER_PORT, "20000");
+
+    // initialize the null master port test properties.
+    sNullMasterPortTestProperties.put(Constants.MASTER_HOSTNAME, "RemoteMaster3");
+
+  }
+
+  @Before
+  public void beforeTests() {
```
Comment (calvinjia) [inline servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java]: I think the convention is to call this method `before`?

**Comment by calvinjia** [comment]
Thanks for the patch! Looks good overall, a few minor comments.

File: servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+
+  @Test
+  public void getMasterAddressTest() {
+    String defaultHostname = NetworkUtils.getLocalHostName(mCustomPropsTachyonConf);
+    int defaultPort = Constants.DEFAULT_MASTER_PORT;
+
+    InetSocketAddress masterAddress = BlockWorkerUtils.getMasterAddress(mCustomPropsTachyonConf);
+    Assert.assertTrue(masterAddress != null);
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java]: Use `assertNotNull`?

File: servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    Assert.assertTrue(masterAddress.equals(new InetSocketAddress("RemoteMaster1", 10000)));
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java]: Use `assertEquals`?

File: servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java (hunk 4 above)
Comment (apc999) [inline servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java]: Same comments apply to line 81, 85 and 89

File: servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java (hunk 5 above)
Comment (apc999) [inline servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java]: Same comments apply to line 82 86 and 90

File: servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+  @Before
+  public void beforeTests() {
+    // init TachyonConf
+    mCustomPropsTachyonConf = new TachyonConf(sTestProperties);
+    mNullMasterHostNameTachyonConf = new TachyonConf(sNullMasterHostNameTestProperties);
+    mNullMasterPortTachyonConf = new TachyonConf(sNullMasterPortTestProperties);
+    mNullTachyonConf = new TachyonConf(sNullTestProperties);
+  }
+
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java]: NIT: one extra line here?

**Comment by apc999** [comment]
Thanks for the patch. It looks good in general, with a few minor comments left.

**Comment by Yours-shupeng** [comment]
@calvinjia @apc999 comments fixed.

File: servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  @Test
+  public void getMasterAddressTest() {
+    String defaultHostname = NetworkUtils.getLocalHostName(mCustomPropsTachyonConf);
+    int defaultPort = Constants.DEFAULT_MASTER_PORT;
+
+    InetSocketAddress masterAddress = BlockWorkerUtils.getMasterAddress(mCustomPropsTachyonConf);
+    Assert.assertNotNull(masterAddress);
+    Assert.assertEquals(masterAddress, new InetSocketAddress("RemoteMaster1", 10000));
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/block/BlockWorkerUtilsTest.java]: for `assertEquals`, the first parameter is the expected value and the second parameter is the value to test.

**Comment by calvinjia** [comment]
+1 LGTM

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Pull request merged**